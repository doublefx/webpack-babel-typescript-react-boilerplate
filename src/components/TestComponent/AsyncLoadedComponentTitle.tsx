import * as React from "react";

const AsyncLoadedComponentTitle = () => <>App Component</>;

// Seems that when using a pure Fragment component, Webpack or Babel does not
// properly include React as a dependency for this file. Referencing React in
// some way forces the inclusion and so the split module will load properly.
React.version;

export default AsyncLoadedComponentTitle;
