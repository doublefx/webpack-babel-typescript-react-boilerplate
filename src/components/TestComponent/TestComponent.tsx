import * as React from "react";
import styled, { withProps } from "styled";

import * as smallImage from "./assets/pexels-photo-270348.small.jpeg";
import * as largeImage from "./assets/pexels-photo-270348.jpeg";

// Making sure withProps is available (correct package aliasing).
export const SecondsCounterText = withProps<{ bold: boolean }>()(styled.span)`
  color: ${props => props.theme.testComponentColor};
  font-weight: ${props => (props.bold ? "600" : "400")};

  ${props => props.theme.media.min.tablet`
    font-size: 24px;
  `}
`;

class TestComponent extends React.Component {
  state = { elapsed: 0, ComponentTitle: null as React.ComponentType | null };

  private interval: number;

  componentDidMount() {
    this.interval = window.setInterval(() => {
      // requestAnimationFrame - verify that it is supported in Jest environment
      requestAnimationFrame(() => {
        this.setState({ elapsed: this.state.elapsed + 1 });
        // console.log("tick");
      });
    }, 1000);

    // async import - verify async loaded component loads correctly
    import(/* webpackChunkName: "asyncLoadedComponentTitle" */ "./AsyncLoadedComponentTitle").then(
      mod => {
        // console.log("### mod ###", mod);
        this.setState({ ComponentTitle: mod.default });
      },
    );
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    /* tslint:disable:strict-boolean-expressions */
    return (
      <>
        {this.state.ComponentTitle && (
          <>
            <this.state.ComponentTitle />
            <br />
          </>
        )}
        <SecondsCounterText bold>
          Seconds: {this.state.elapsed}
        </SecondsCounterText>
        <br />
        <img src={smallImage} style={{ width: 100 }} />
        <br />
        <img src={largeImage} style={{ width: 100 }} />
        <br />
        <p>NODE_ENV: {process.env.NODE_ENV}</p>
      </>
    );
    /* tslint:enable:strict-boolean-expressions */
  }
}

export default TestComponent;
