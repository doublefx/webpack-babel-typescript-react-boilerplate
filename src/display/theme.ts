const theme = {
  testComponentColor: "#eee",
};

export type Theme = typeof theme;

export default theme;
