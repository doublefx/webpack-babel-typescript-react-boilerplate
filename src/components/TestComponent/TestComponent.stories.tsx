import * as React from "react";
import { storiesOf } from "@storybook/react";
import TestComponent from "./TestComponent";

storiesOf("BoilerplateTestComponent", module).add("test render", () => (
  <TestComponent />
));
