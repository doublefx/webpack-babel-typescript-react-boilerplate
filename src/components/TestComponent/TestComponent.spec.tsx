import * as React from "react";
import * as renderer from "react-test-renderer";
import { ThemeProvider } from "display";
import TestComponent from "./TestComponent";

const createRenderer = () =>
  renderer.create(
    <ThemeProvider>
      <TestComponent />
    </ThemeProvider>,
  );

it("renders loads subcomponent asynchronously", done => {
  const component = createRenderer();

  // Renders component without subcomponent <AsyncLoadedComponentTitle />
  expect(component.toJSON()).toMatchSnapshot();

  setTimeout(() => {
    // Renders asynchronously loaded subcomponent
    expect(component.toJSON()).toMatchSnapshot();

    setTimeout(() => {
      // Increments timer count
      expect(component.toJSON()).toMatchSnapshot();
      done();
    }, 1000);
  }, 100);

});


function doubleSay (str:string) :string {
    return str + ", " + str;
}
function capitalize (str:string):string {
    return str[0].toUpperCase() + str.substring(1);
}
function exclaim (str:string):string {
    return str + '!';
}
/*  let result:string = exclaim(capitalize(doubleSay("hello")));
    result //=> "Hello, hello!"*/

it("understands pipe operator", () => {
    const test:string = "hello"
        |> doubleSay
        |> capitalize
        |> exclaim;

    expect(test).toEqual("Hello, hello!");
});