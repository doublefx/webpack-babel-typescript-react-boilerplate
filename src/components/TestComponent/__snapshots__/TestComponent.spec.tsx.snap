// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loads subcomponent asynchronously 1`] = `
Array [
  .c0 {
  color: #eee;
  font-weight: 600;
}

@media (min-width:768px) {
  .c0 {
    font-size: 24px;
  }
}

<span
    className="c0"
  >
    Seconds: 
    0
  </span>,
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
  <br />,
  <p>
    NODE_ENV: 
    test
  </p>,
]
`;

exports[`renders loads subcomponent asynchronously 2`] = `
Array [
  "App Component",
  <br />,
  .c0 {
  color: #eee;
  font-weight: 600;
}

@media (min-width:768px) {
  .c0 {
    font-size: 24px;
  }
}

<span
    className="c0"
  >
    Seconds: 
    0
  </span>,
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
  <br />,
  <p>
    NODE_ENV: 
    test
  </p>,
]
`;

exports[`renders loads subcomponent asynchronously 3`] = `
Array [
  "App Component",
  <br />,
  .c0 {
  color: #eee;
  font-weight: 600;
}

@media (min-width:768px) {
  .c0 {
    font-size: 24px;
  }
}

<span
    className="c0"
  >
    Seconds: 
    1
  </span>,
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
  <br />,
  <img
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />,
  <br />,
  <p>
    NODE_ENV: 
    test
  </p>,
]
`;
